services/ui-src/src/components/cards/EntityCard.tsx

```diff
@@ -71,51 +37,90 @@ export const EntityCard = ({
         </Heading>
-        <Text sx={sx.description}>{data.standardDescription}</Text>
+        <Text sx={sx.description}>
+          {formattedEntityData.standardDescription}
+        </Text>
         <Text sx={sx.subtitle}>General category</Text>
-        <Text sx={sx.subtext}>{data.standardType}</Text>
+        <Text sx={sx.subtext}>{formattedEntityData.standardType}</Text>
         <Button
           variant="outline"
           size="sm"
-          sx={sx.editEntityButton}
+          sx={sx.editButton}
+          data-testid="editEntityButton"
           leftIcon={<Image src={editIcon} alt="edit icon" height="1rem" />}
           onClick={() => openAddEditEntityModal(entity)}
         >
-          Edit measure
-        </Button>
-        <Text sx={sx.unfinishedMessage}>
-          Complete the remaining indicators for this access measure by entering
-          details.
-        </Text>
-        <Button size="sm" sx={sx.enterDrawerButton}>
-          Enter details
+          {dashboard.editEntityButtonText}
         </Button>
+        {entityCompleted ? (
+          <>
+            <Flex sx={sx.highlightContainer}>
+              <Box sx={sx.highlightSection}>
+                <Text sx={sx.subtitle}>Provider</Text>
+                <Text sx={sx.subtext}>{formattedEntityData?.provider}</Text>
+              </Box>
+              <Box sx={sx.highlightSection}>
+                <Text sx={sx.subtitle}>Region</Text>
+                <Text sx={sx.subtext}>{formattedEntityData?.region}</Text>
+              </Box>
+              <Box sx={sx.highlightSection}>
+                <Text sx={sx.subtitle}>Population</Text>
+                <Text sx={sx.subtext}>{formattedEntityData?.population}</Text>
+              </Box>
+            </Flex>
+            <Text sx={sx.subtitle}>Monitoring Methods</Text>
+            <Text sx={sx.subtext}>
+              {formattedEntityData?.monitoringMethods.join(", ")}
+            </Text>
+            <Text sx={sx.subtitle}>Frequency of oversight methods</Text>
+            <Text sx={sx.subtext}>{formattedEntityData.methodFrequency}</Text>
+          </>
+        ) : (
+          <Text sx={sx.unfinishedMessage}>
+            Complete the remaining indicators for this access measure by
+            entering details.
+          </Text>
+        )}
+        {entityCompleted ? (
+          <Button
+            variant="outline"
+            size="sm"
+            sx={sx.editButton}
+            data-testid="editDetailsButton"
+            leftIcon={<Image src={editIcon} alt="edit icon" height="1rem" />}
+            onClick={() => openDrawer(entity)}
+          >
+            {dashboard.editEntityDetailsButtonText}
+          </Button>
+        ) : (
+          <Button
+            size="sm"
+            sx={sx.openDrawerButton}
+            data-testid="enterDetailsButton"
+            onClick={() => openDrawer(entity)}
+          >
+            Enter details
+          </Button>
+        )}
```

Comment:
could be combined with the above, though I think you left this separate intentionally to make it a little cleaner?

Reply:
yeah i thought about it initially but seemed like it would be too messy looking. gave it another shot in this commit: https://github.com/CMSgov/mdct-mcr/pull/4247/commits/4f950d10380ef6ee7c45fd547f0786b8bca8c982

i think you're right that it looks better this way.